fast64_internal/oot/c_writer/oot_scene_bootup.py

```diff
@@ -0,0 +1,137 @@
+import os, re, bpy
+import string
+from bpy.utils import register_class, unregister_class
+from ...utility import PluginError, writeFile
+from ..oot_constants import ootEnumHeaderMenuComplete
+from typing import Callable, Iterable, Any, List
+
+
+def setBootupScene(exportPath: str, entranceIndex: str, options: "OOTBootupSceneOptions"):
+
+    linkAge = "LINK_AGE_CHILD"
+    timeOfDay = "NEXT_TIME_NONE"
+    cutsceneIndex = "0xFFEF"
+    saveFileNameData = ", ".join(["0x" + format(i, "02X") for i in stringToSaveNameBytes(options.newGameName)])
+    if options.overrideHeader:
+        timeOfDay, linkAge = getParamsFromOptions(options)
+        if options.headerOption == "Cutscene":
+            cutsceneIndex = "0xFFF" + format(options.cutsceneIndex - 4, "X")
+
+    data = (
+        f"#ifndef CONFIG_BOOTUP_H\n"
+        + f"#define CONFIG_BOOTUP_H\n\n"
+        + (("" if options.bootMode == "Play" else "// ") + "#define BOOT_TO_SCENE\n")
+        + (("" if options.newGameOnly else "// ") + "#define BOOT_TO_SCENE_NEW_GAME_ONLY\n")
+        + (("" if options.bootMode == "File Select" else "// ") + "#define BOOT_TO_FILE_SELECT\n")
+        + f"#define BOOT_ENTRANCE {entranceIndex}\n"
+        + f"#define BOOT_AGE {linkAge}\n"
+        + f"#define BOOT_TIME {timeOfDay}\n"
+        + f"#define BOOT_CUTSCENE {cutsceneIndex}\n"
+        + f"#define BOOT_PLAYER_NAME {saveFileNameData}\n\n"
+        + f"#endif\n"
+    )
+
+    writeFile(os.path.join(exportPath, "include/config/config_bootup.h"), data)
+
+
+def clearBootupScene(exportPath: str):
+
+    data = (
+        f"#ifndef CONFIG_BOOTUP_H\n"
+        + f"#define CONFIG_BOOTUP_H\n\n"
+        + f"// #define BOOT_TO_SCENE\n"
+        + f"// #define BOOT_TO_SCENE_NEW_GAME_ONLY\n"
+        + f"// #define BOOT_TO_FILE_SELECT\n"
+        + f"#define BOOT_ENTRANCE 0\n"
+        + f"#define BOOT_AGE LINK_AGE_CHILD\n"
+        + f"#define BOOT_TIME NEXT_TIME_NONE\n"
+        + f"#define BOOT_CUTSCENE 0xFFEF\n"
+        + f"#define BOOT_PLAYER_NAME 0x15, 0x12, 0x17, 0x14, 0x3E, 0x3E, 0x3E, 0x3E\n\n"
+        + f"#endif\n"
+    )
+
+    writeFile(os.path.join(exportPath, "include/config/config_bootup.h"), data)
+
+
+def getParamsFromOptions(options: Any) -> tuple[str, str]:
+    timeOfDay = (
+        "NEXT_TIME_DAY"
+        if options.headerOption == "Child Day" or options.headerOption == "Adult Day"
+        else "NEXT_TIME_NIGHT"
+    )
+
+    linkAge = (
+        "LINK_AGE_ADULT"
+        if options.headerOption == "Adult Day" or options.headerOption == "Adult Night"
+        else "LINK_AGE_CHILD"
+    )
+
+    return timeOfDay, linkAge
+
+
+# converts ascii text to format for save file name.
+# see src/code/z_message_PAL.c:Message_Decode()
+def stringToSaveNameBytes(name: str) -> bytearray:
```

Comment:
One little thing to note here is that soon I intend to change it to just use ascii in hackeroot, but this is good for now.